### PR TITLE
Update deprecated.jl

### PR DESCRIPTION
**Pull request by matrixmorpheus**
Some rules can be bent. #9803

Others can be broken. #9779

### CODE REVIEW

File: base/deprecated.jl [hunk 1]
```diff
@@ -273,3 +273,12 @@ const base64 = base64encode
 
 @deprecate error(ex::Exception) throw(ex)
 @deprecate error{E<:Exception}(::Type{E}) throw(E())
+
+@deprecate CholmodSparse!(colpt, rowval, nzval, m) CholmodSparse(colpt, rowval, nzval, m)
+@ddprecate lufact!(S::SparseMatrixCSC) lufact(S)
```
Comment (andreasnoack) [inline base/deprecated.jl]: Typo in `@ddprecate`.

**Comment by andreasnoack** [comment]
Thanks for doing this.

**Comment by ViralBShah** [comment]
Love the matrix quotes!

**Comment by jiahao** [comment]
It looks like the `@deprecate` macro cannot handle unexported symbols. See [error in log](https://travis-ci.org/JuliaLang/julia/jobs/47378382#L3077):

```
invalid usage of @deprecate
```

Offending line:

```
 @deprecate Triangular{T,S,:U,true} Base.LinAlg.UnitUpperTriangular
```

**Comment by andreasnoack** [comment]
I've made a small change in https://github.com/JuliaLang/julia/commit/e40c11b2cd584340b436e73352d81898b68fb41e to avoid a vast amount of ambiguities on packages. Hence `TriangularUnion` should be `AbstractTriangular`.

**Comment by ivarne** [comment]
It's kind of sad to not have deprecations when we change something. What should we do?

**Comment by jiahao** [comment]
bump

**Comment by andreasnoack** [comment]
Isn't this one depending on #9830?

**Comment by jiahao** [comment]
Oh, right.

**Comment by ViralBShah** [comment]
Bump. What do we do here now, with the recent update to Cholmod?

**Comment by jiahao** [comment]
This PR needs to be redone and we still need #9830.

**Comment by ViralBShah** [comment]
I am not sure that the deprecated `lufact!(s::SparseMatrixCSC)` should call `lufact` - although there is no harm in doing so. My guess is that due to its buggy nature, it would not have been used much at all. `CholmodSparse` is/was an internal API, and we don't need to worry about it.

Cc: @andreasnoack on what to do with the triangular stuff. Maybe best to just close this one.

**Pull request closed without merging**